docs/setup/install-new.md

```diff
@@ -47,23 +49,60 @@ Click on the **Deploy to Azure** icon below to begin the infrastructure deployme
-  - paste the `appId` vaule in the `Aks Service Principal App Id` template field
-  - paste the `password` value in the `Aks Service Principal Client Secret` field
-  - paste the `objectId` value in the `Aks Service Principal Object Id` field
+1. Download the ARM template and modify the template as needed.
+    ```bash
+    wget https://raw.githubusercontent.com/Azure/application-gateway-kubernetes-ingress/master/deploy/azuredeploy.json -O template.json
+    ```
 
-Note: To deploy an **RBAC** enabled cluster, set the `aksEnabledRBAC` field to `true`
+1. Create the parameter file.
+    ```bash
+    cat <<EOF > parameters.json
+    {
+      "aksServicePrincipalAppId": { "value": "$appId" },
+      "aksServicePrincipalClientSecret": { "value": "$password" },
+      "aksServicePrincipalObjectId": { "value": "$objectId" },
+      "aksEnableRBAC": { "value": false }
+    }
+    EOF
+    ```
 
-<a href="https://portal.azure.com/#create/Microsoft.Template/uri/https%3A%2F%2Fraw.githubusercontent.com%2FAzure%2Fapplication-gateway-kubernetes-ingress%2Fmaster%2Fdeploy%2Fazuredeploy.json" target="_blank">
-<img src="http://azuredeploy.net/deploybutton.png"/>
-</a>
+    #### Important
+    Please use the `appId`, `objectId`, and `password` values from the `az` commands above and
+    paste them in the corresponding ARM template fields:
+      - paste the `appId` vaule in the `Aks Service Principal App Id` template field
+      - paste the `password` value in the `Aks Service Principal Client Secret` field
+      - paste the `objectId` value in the `Aks Service Principal Object Id` field
 
-Navigate to the deployment output and record the parameters:
-[Azure portal](https://portal.azure.com/): `Home -> *resource group* -> Deployments -> *new deployment* -> Outputs`)
+    Note: To deploy an **RBAC** enabled cluster, set the `aksEnabledRBAC` field to `true`
+
+1. Deploy the ARM template.
+    ```bash
+    resourceGroupName="MyResourceGroup"
+    location="westus2"
+    deploymentName="ingress-appgw"
+
+    # create a resource group
+    az group create -n $resourceGroupName -l $location
+
+    # modify the template as needed
+    az group deployment create \
+            -g $resourceGroupName \
+            -n $deploymentName \
+            --template-file template.json \
+            --parameters ./parameters.json
+    ```
+
+1. Save the output from the deployment
+    ```bash
+    az group deployment show -g $resourceGroupName -n $deploymentName --query "properties.outputs" -o json > outputs.json
+    aksClusterName=$(jq -r ".aksClusterName.value" outputs.json)
+    applicationGatewayName=$(jq -r ".applicationGatewayName.value" outputs.json)
+    resourceGroupName=$(jq -r ".resourceGroupName.value" outputs.json)
+    subscriptionId=$(jq -r ".subscriptionId.value" outputs.json)
+    identityClientId=$(jq -r ".identityClientId.value" outputs.json)
+    identityResourceId=$(jq -r ".identityResourceId.value" outputs.json)
```

Comment:
It might be good to separate these bash var creation from the first line.  One is dump the output. The second blob is - create variables.

We can even move this block of code right above the creation of the helm config - so it is clear where these are used.